tests/test_integration.py

```diff
@@ -89,10 +91,48 @@ def test_good_result_manifests():
     """
     result = run_query("test_all_result_groups", Namespace())
 
-    for group, count in result["data"]:
+    for group, _ in result["data"]:
         if group is None:
             continue
 
         assert (
             not task.is_bad_group("x", group) and "\\" not in group
         ), f"{group} group is bad!"
+
+
+def test_caching_of_tasks(tmp_path):
```

Comment:
Nit: You could also write a unit test, by mocking the answers from Taskcluster via responses

Reply:
Do you have any trick on how to collect all the responses?
One thing I've always struggled with mocking is collecting all the URLs and the individual responses.

Reply:
I decided we can get this landed and see if we want to push for it on a follow up if that's acceptable to you.